### PR TITLE
Install cidmaps by default

### LINKED ISSUE #4350: cidmaps should be bundled
Fontforge repository have https://github.com/fontforge/fontforge/tree/master/contrib/cidmap where cidmap files are available. 
I got below questions about them.
1)  I am not sure why those cidmap files are not available to be installed using `-DENABLE_MAINTAINER_TOOLS=ON` cmake option.
2)  Can some of them be updated? 
https://github.com/adobe-type-tools/Adobe-CNS1 shows Adobe-CNS1-7.cidmap is available.
https://github.com/adobe-type-tools/Adobe-KR shows Adobe-KR1-9.cidmap is available.

### LINKED ISSUE #4411: "Could not open CIDmap file" - Adobe-Japan1-3
* Provide a general summary of the issue in the **Title** above.  
* Before you open an issue, please check if a similar issue already exists or has been closed before.

### When reporting a bug/issue:
- [ ] Screenshot
- [x] The FontForge version and the operating system you're using
20200314 on Windows 10 64bit
- [x] The behavior you expect to see, and the actual behavior
- [x] Steps to reproduce the behavior 
Opening and editing a Japanese font with Adobe-Japan1-3 CIDmap. I don't know what's wrong with it, but FontForge tells me that CIDmap files are missing. I download the CIDmap files from the indicated place, and let FontForge load it, but it gives me the "Could not open CIDmap file" error message.
- [ ] \(optional) Possible solution/fix/workaround

### PR DESCRIPTION
**Pull request by ctrlcctrlv**
There's no reason not to do this. We already distribute them as part of
the repository.

* Closes #4411.
* Closes #4350.
* Related also to #1831. After we make it so we're installing the
  cidmaps in the repo, it would make sense to assure that they're
  actually the most up-to-date ones.

### Type of change

- **Bug fix**
- **Non-breaking change**

### CODE REVIEW

File: fontforge/encoding.c [hunk 1]
```diff
@@ -1155,9 +1157,7 @@ return( maybe );
 	buf = smprintf( "%s-%s-*.cidmap", registry, ordering );
 	if ( maybe==NULL && maybefile==NULL ) {
 	    buts3[0] = _("_Browse"); buts3[1] = _("_Give Up"); buts3[2] = NULL;
-	    ret = ff_ask(_("No cidmap file..."),(const char **)buts3,0,1,_("FontForge was unable to find a cidmap file for this font. It is not essential to have one, but some things will work better if you do. If you have not done so you might want to download the cidmaps from:\n   http://FontForge.sourceforge.net/cidmaps.tgz\nand then gunzip and untar them and move them to:\n  %.80s\n\nWould you like to search your local disk for an appropriate file?"),
-		    getFontForgeShareDir()==NULL?"/usr/share/fontforge":getFontForgeShareDir()
```
Comment (frank-trampe) [inline fontforge/encoding.c:1159]: Why drop the path?

File: fontforgeexe/startui.c [hunk 2]
```diff
@@ -1080,12 +1080,11 @@ int fontforge_main( int argc, char **argv ) {
     textdomain("FontForge");
     GResourceUseGetText();
     {
-	char path[PATH_MAX];
-	snprintf(path, PATH_MAX, "%s%s", getShareDir(), "/pixmaps" );
-	GGadgetSetImageDir( path );
+	GGadgetSetImageDir( getPixmapDir() );
```
Comment (frank-trampe) [inline fontforgeexe/startui.c:1083]: GGadgetSetImageDir leaves its argument untouched. I think that we need to free it here.

**Review by jtanx — CHANGES_REQUESTED** [review]
These are the changes I would make


```patch
diff --git a/contrib/cidmap/CMakeLists.txt b/contrib/cidmap/CMakeLists.txt
index 709a4de6c..88f20e0fd 100644
--- a/contrib/cidmap/CMakeLists.txt
+++ b/contrib/cidmap/CMakeLists.txt
@@ -1,22 +1,29 @@
 # Distributed under the original FontForge BSD 3-clause license
 
 if(ENABLE_MAINTAINER_TOOLS)
-    add_executable(mapcns1 src/mapcns1.c)
-    add_executable(mapgb1 src/mapgb1.c)
-    add_executable(mapjapan1 src/mapjapan1.c)
-    add_executable(mapjapan2 src/mapjapan2.c)
-    add_executable(mapkorean src/mapkorean.c)
+  add_executable(mapcns1 src/mapcns1.c)
+  add_executable(mapgb1 src/mapgb1.c)
+  add_executable(mapjapan1 src/mapjapan1.c)
+  add_executable(mapjapan2 src/mapjapan2.c)
+  add_executable(mapkorean src/mapkorean.c)
 
-    target_link_libraries(mapcns1 PRIVATE fontforge)
-    target_link_libraries(mapgb1 PRIVATE fontforge)
-    target_link_libraries(mapjapan1 PRIVATE fontforge)
-    target_link_libraries(mapjapan2 PRIVATE fontforge)
-    target_link_libraries(mapkorean PRIVATE fontforge)
+  target_link_libraries(mapcns1 PRIVATE fontforge)
+  target_link_libraries(mapgb1 PRIVATE fontforge)
+  target_link_libraries(mapjapan1 PRIVATE fontforge)
+  target_link_libraries(mapjapan2 PRIVATE fontforge)
+  target_link_libraries(mapkorean PRIVATE fontforge)
 endif()
 
-file(GLOB_RECURSE cidmaps *.cidmap)
-
 install(
-    FILES ${cidmaps}
-    DESTINATION ${CMAKE_INSTALL_DATADIR}/fontforge/cidmap
+  FILES
+    Adobe-CNS1-6.cidmap
+    Adobe-GB1-5.cidmap
+    Adobe-Identity-0.cidmap
+    Adobe-Japan1-5.cidmap
+    Adobe-Japan1-6.cidmap
+    Adobe-Japan1-7.cidmap
+    Adobe-Japan2-0.cidmap
+    Adobe-Korea1-2.cidmap
+  DESTINATION ${CMAKE_INSTALL_DATADIR}/fontforge/cidmap
+  COMPONENT cidmaps
 )
-- 
2.32.0

```


```patch
diff --git a/fontforge/encoding.c b/fontforge/encoding.c
index 0f99dda7f..e22caf958 100644
--- a/fontforge/encoding.c
+++ b/fontforge/encoding.c
@@ -1095,7 +1095,7 @@ return( ret );
 struct cidmap *FindCidMap(char *registry,char *ordering,int supplement,SplineFont *sf) {
     struct cidmap *map, *maybe=NULL;
     char *file;
-    char *maybefile=NULL;
+    char *maybefile=NULL, *sharedir=NULL;
     int maybe_sup = -1;
     const char *buts[3], *buts2[3], *buts3[3];
     char *buf = NULL;
@@ -1117,8 +1117,8 @@ return( map );
 return( maybe );	/* User has said it's ok to use maybe at this supplement level */
 
     file = SearchDirForCidMap(".",registry,ordering,supplement,&maybefile);
-    char* sharedir = getShareSubDir("/cidmap");
     if ( file==NULL ) {
+	sharedir = getShareSubDir("/cidmap");
 	file = SearchDirForCidMap(sharedir,registry,ordering,supplement,&maybefile);
     }
 
@@ -1157,7 +1157,8 @@ return( maybe );
 	buf = smprintf( "%s-%s-*.cidmap", registry, ordering );
 	if ( maybe==NULL && maybefile==NULL ) {
 	    buts3[0] = _("_Browse"); buts3[1] = _("_Give Up"); buts3[2] = NULL;
-	    ret = ff_ask(_("No cidmap file..."),(const char **)buts3,0,1,_("FontForge was unable to find a cidmap file for this font. It is not essential to have one, but some things will work better if you do. Would you like to search your local disk for an appropriate file?"));
+	    ret = ff_ask(_("No cidmap file..."),(const char **)buts3,0,1,_("FontForge was unable to find a cidmap file for this font. The following path was searched:\n   %.80s\n\nIt is not essential to have one, but some things will work better if you do. Would you like to search your local disk for an appropriate file?"),
+	    sharedir);
 	    if ( ret==1 || no_windowing_ui )
 		buf = NULL;
 	}
diff --git a/fontforgeexe/startui.c b/fontforgeexe/startui.c
index 77f1080e1..cddf149a8 100644
--- a/fontforgeexe/startui.c
+++ b/fontforgeexe/startui.c
@@ -1080,11 +1080,13 @@ int fontforge_main( int argc, char **argv ) {
     textdomain("FontForge");
     GResourceUseGetText();
     {
-	GGadgetSetImageDir( getPixmapDir() );
+	char* path = getPixmapDir();
+	GGadgetSetImageDir( path );
+    free(path);
 
-	char* resource_filename = getShareSubDir("/resources/fontforge.resource");
-	GResourceAddResourceFile(resource_filename, GResourceProgramName,false);
-	free(resource_filename);
+	path = getShareSubDir("/resources/fontforge.resource");
+	GResourceAddResourceFile(path, GResourceProgramName,false);
+	free(path);
     }
     hotkeysLoad();
 //    loadPrefsFiles();
diff --git a/gutils/fsys.c b/gutils/fsys.c
index 3f8411ccf..a3689cb14 100644
--- a/gutils/fsys.c
+++ b/gutils/fsys.c
@@ -49,7 +49,7 @@
 
 static char *program_dir = NULL;
 static char dirname_[MAXPATHLEN+1];
-static char *FONTFORGE_SHARE_SUBDIR = "/share/fontforge";
+static const char *FONTFORGE_SHARE_SUBDIR = "/share/fontforge";
 
 /**
  * \brief Removes the extension from a file path, if it exists.
@@ -835,7 +835,7 @@ void FindProgDir(char *prog) {
 // When calling this function, make sure you prepend "/" to the subdirectory
 // you want. So, to get e.g. /usr/share/cidmaps, call
 // getShareSubDir("/cidmaps").
-char *getShareSubDir(char* subdir) {
+char *getShareSubDir(const char* subdir) {
     char *sharedir, *pt;
     int len;
 
diff --git a/inc/gfile.h b/inc/gfile.h
index dddec327c..d09a5e729 100644
--- a/inc/gfile.h
+++ b/inc/gfile.h
@@ -98,7 +98,7 @@ extern char *GFileReadAll(char *name);
 extern int   GFileWriteAll(char *filepath, char *data);
 extern void  FindProgDir(char *prog);
 extern char *getShareDir();
-extern char *getShareSubDir(char* subdir);
+extern char *getShareSubDir(const char* subdir);
 extern char *getLocaleDir(void);
 extern char *getPixmapDir(void);
 extern char *getHelpDir(void);
-- 
2.32.0

```

CMake changes were to keep in style with the rest (kitware style/2 space indents) and to not use globs.

Code changes just addressing a couple of the comments from Frank.

**Comment by pnemade** [comment]
Can we have some progress here? Is this ready to be merged?

**Comment by pnemade** [comment]
Thank you @jtanx

**Pull request merged**